pkg/gateway/dispatcher.go

```diff
@@ -372,7 +372,7 @@ func (d *Dispatcher) buildSnapshot(ctx context.Context) {
 		bs, _ := json.MarshalIndent(snapshot, "", "  ")
 		dlog.Errorf(ctx, "Dispatcher Snapshot inconsistency: %v: %s", err, bs)
 	} else {
-		d.snapshot = &snapshot
+		d.snapshot = snapshot
```

Comment:
is this due to an api change in the go-control-plane? Seems odd to switch from passing by reference to passing by value

Reply:
It is passing the pointer still the call to `NewSnapshot` which generated this snapshot had its return type changed as you can see based on this git blame.

https://github.com/envoyproxy/go-control-plane/blame/a51ba6fa54b2e107f5606a0e2799f8afed116c14/pkg/cache/v3/snapshot.go

It was:
`func NewSnapshot(version string, resources map[resource.Type][]types.Resource) (Snapshot, error)`
and is now
`func NewSnapshot(version string, resources map[resource.Type][]types.Resource) (*Snapshot, error)`